### DISCUSSION TITLE
ssh-agent integration

### DISCUSSION THREAD

**Discussion by nullishamy** [nushell/nushell#9192] [General]
### Question

Looking to move to nushell from zsh (love it so far!), missing some [ssh agent integration](https://github.com/ohmyzsh/ohmyzsh/blob/master/plugins/ssh-agent/ssh-agent.plugin.zsh), how would you achieve similar functionality in nushell (or is there an existing solution to use)?

**Comment by rgwood** [comment]
What functionality are you looking for, exactly?

**Comment by amtoine** [comment]
@nullishamy 
in my config i activate an `ssh` agent with the following snippet
```bash
# start the ssh agent to allow SSO with ssh authentication
# very usefull with `github` over the ssh protocol
#
# see https://www.nushell.sh/cookbook/misc.html#manage-ssh-passphrases
#
let-env SSH_AGENT_TIMEOUT = 300
ssh-agent -c -t $env.SSH_AGENT_TIMEOUT
| lines
| first 2
| parse "setenv {name} {value};"
| transpose -i -r -d
| load-env
```
is that what you're looking for? :yum:

**Reply by amtoine** [reply]
which is simply taken and tweaked a little from the [*Manage SSH passphrases* chapter](https://www.nushell.sh/cookbook/misc.html#manage-ssh-passphrases) of the cookbook :relieved:

**Reply by nullishamy** [reply]
Yep, that's exactly what I was looking for, thanks!

**Reply by amtoine** [reply]
anytime :blush:
